src/interceptor/utils/cloneObject.ts

```diff
@@ -0,0 +1,36 @@
+import { pinoLogger } from '../../logger';
+
+const logger = pinoLogger.child({ module: 'cloneObject' });
+
+function isPlainObject(obj?: Record<string, any>): boolean {
+  logger.debug('is plain object?', obj);
+
+  if (obj == null || !obj.constructor?.name) {
+    logger.debug('given object is undefined, not a plain object...');
+    return false;
+  }
+
+  logger.debug('checking the object constructor:', obj.constructor.name);
+  return obj.constructor.name === 'Object';
+}
+
+export function cloneObject<ObjectType extends Record<string, any>>(
+  obj: ObjectType
+): ObjectType {
+  logger.debug('cloning object:', obj);
```

Comment:
can you check and/or remove these debugging statements if the payloads are getting stored into memory regardless of whether debug is turned on or not?

Reply:
I disabled it for now. I left the logs to allow local debugging, so they shouldn't affect production builds